### PR TITLE
feat(format/html): port `JsxChildList` formatting to `HtmlElementList`

### PR DESCRIPTION
**Pull request by dyc3**
## Summary

Given the similarity of JSX and HTML, I figured it would save a lot of time to simply port over the applicable portions of JSX formatting to HTML.

This PR attempts to do so for `JsxChildList` -> `HtmlElementList`.

## Test Plan

`quick_test` should run without panicking.

### CODE REVIEW

File: crates/biome_html_formatter/src/utils/children.rs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+/// ```
+pub fn is_meaningful_html_text(text: &str) -> bool {
+    let mut has_newline = false;
+    for c in text.chars() {
+        // If there is a non-whitespace character
+        if !HTML_WHITESPACE_CHARS.contains(&c) {
+            return true;
+        } else if c == '\n' {
+            has_newline = true;
+        }
```
Comment (chansuke) [inline crates/biome_html_formatter/src/utils/children.rs:38]: Is it useful to add a comment explaining the newline detection?

```suggestion
        } else if c == '\n' {
            // A newline is detected. This is tracked separately, 
            // potentially to differentiate text with just line breaks from other whitespace.
            has_newline = true;
        }
```

File: crates/biome_html_formatter/src/utils/children.rs [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+    I: IntoIterator<Item = AnyHtmlElement>,
+{
+    let mut builder = HtmlSplitChildrenBuilder::new();
+
+    for child in children {
+        match child {
+            AnyHtmlElement::HtmlContent(text) => {
+                // Split the text into words
+                // Keep track if there's any leading/trailing empty line, new line or whitespace
+
```
Comment (chansuke) [inline crates/biome_html_formatter/src/utils/children.rs:175]: Is this space intentional?

Comment (dyc3) [reply crates/biome_html_formatter/src/utils/children.rs:175]: This is mostly copied strait from the formatter for `JsxChildList`.

File: crates/biome_html_formatter/src/utils/children.rs (hunk 1 above)
Comment (dyc3) [reply crates/biome_html_formatter/src/utils/children.rs:38]: This is explained in the doc comment for the function.

**Comment by dyc3** [comment]
After some further testing, this code does not handle comments very well. Marking as draft for now.

**Comment by dyc3** [comment]
I think I've resolved the comments thing in other PRs. I've tested this with the prettier test suite, and it gets most of the way there in terms of compatibility.

File: crates/biome_html_formatter/src/comments.rs [hunk 3]
```diff
@@ -82,8 +97,14 @@ impl CommentStyle for HtmlCommentStyle {
...
         &self,
-        _comment: DecoratedComment<Self::Language>,
+        comment: DecoratedComment<Self::Language>,
     ) -> CommentPlacement<Self::Language> {
-        todo!();
+        // dbg!(&comment);
+        // let Some(following_node) = comment.following_node() else {
+        //     return CommentPlacement::Default(comment);
+        // };
+        // CommentPlacement::leading(following_node.clone(), comment)
```
Comment (ematipico) [inline crates/biome_html_formatter/src/comments.rs:106]: To remove

File: crates/biome_html_formatter/src/html/lists/element_list.rs [hunk 4]
```diff
@@ -1,10 +1,720 @@
...
+}
+
+#[derive(Debug)]
+pub(crate) enum FormatChildrenResult {
+    ForceMultiline(FormatMultilineChildren),
+    BestFitting {
+        flat_children: FormatFlatChildren,
+        expanded_children: FormatMultilineChildren,
+    },
+}
```
Comment (ematipico) [inline crates/biome_html_formatter/src/html/lists/element_list.rs:50]: Should we name this `FormatChildrenLayout` instead? We should also document the variants and explain when they are applied, maybe with a code snippet

**Review by ematipico — APPROVED** [review]

File: crates/biome_html_formatter/src/html/lists/element_list.rs (hunk 4 above)
Comment (dyc3) [reply crates/biome_html_formatter/src/html/lists/element_list.rs:50]: Actually, upon further inspection, we already have the `HtmlChildListLayout` enum, so renaming this to `FormatChildrenLayout` could get confusing.

**Pull request merged**